src/common/types.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ *    Copyright (c) 2020, The OpenThread Authors.
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *    1. Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *    3. Neither the name of the copyright holder nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *    ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *    LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *    CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <arpa/inet.h>
+#include <sys/socket.h>
+
+#include "code_utils.hpp"
+#include "common/types.hpp"
+
+namespace otbr {
+
+Ip6Address::Ip6Address(const otIp6Address &aAddress)
+{
+    static_assert(sizeof(*this) == sizeof(aAddress), "wrong Ip6Address size");
+
+    m32[0] = aAddress.mFields.m32[0];
+    m32[1] = aAddress.mFields.m32[1];
+    m32[2] = aAddress.mFields.m32[2];
+    m32[3] = aAddress.mFields.m32[3];
+}
+
+std::string Ip6Address::ToString() const
+{
+    char strbuf[INET6_ADDRSTRLEN];
+
+    VerifyOrDie(inet_ntop(AF_INET6, this->m8, strbuf, sizeof(strbuf)) != nullptr,
+                "Failed to convert Ip6 address to string");
+
+    return std::string(strbuf);
```

Comment:
It's convenient to use `std::string`. 

We can't return a pointer to `strbuf` since it's on the stack. We may also define a new class like `Ip6AddressString` but I don't think it's a better idea than using `std::string`. 

Thoughts?